tardis/visualization/tools/sdec_plot.py

```diff
@@ -1013,31 +1230,40 @@ def _plot_emission_mpl(self):
                 label="Other elements",
             )
 
-        elements_z = self.elements
         # Contribution from each element
-        for i, atomic_number in enumerate(elements_z):
-            # Add a try catch because elemets_z comes from the total contribution of absorption and emission.
-            # Therefore it's possible that something in elements_z is not in the emission df
+        for ii, identifier in enumerate(self.species):
             try:
                 lower_level = upper_level
                 upper_level = (
                     lower_level
-                    + self.emission_luminosities_df[atomic_number].to_numpy()
+                    + self.emission_luminosities_df[identifier].to_numpy()
                 )
 
                 self.ax.fill_between(
                     self.plot_wavelength,
                     lower_level,
                     upper_level,
-                    color=self.cmap(i / len(self.elements)),
+                    color=self._color_list[ii],
                     cmap=self.cmap,
                     linewidth=0,
                 )
             except:
-                print(
-                    atomic_number2element_symbol(atomic_number)
-                    + " is not in the emitted packets; skipping"
-                )
+                # Add notifications that this species was not in the emission df
+                if self._species_list is None:
+                    longer.info(
+                        atomic_number2element_symbol(identifier)
+                        + " is not in the emitted packets; skipping"
+                    )
+                else:
+                    # Get the ion number and atomic number for each species
+                    ion_number = identifier % 100
+                    atomic_number = (identifier - ion_number) / 100
+
+                    logger.info(
+                        atomic_number2element_symbol(atomic_number)
+                        + int_to_roman(ion_number + 1)
+                        + " is not in the emitted packets; skipping"
```

Comment:
```suggestion
                        f"{atomic_number2element_symbol(atomic_number)"
                        f"{int_to_roman(ion_number + 1)}"
                        f" is not in the emitted packets; skipping"
```
And at multiple places in this file